### PR TITLE
ASC-705 Add unit tests for delete_instance

### PR DESCRIPTION
**Pull request by johnduarte**
This commit adds unit tests for the `delete_instance` helper.

### CODE REVIEW

File: tests/helpers/test_delete_instance.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+def test_successful_deletion(mocker):
+    """Verify delete_instance returns None when server instance is
+    successfully deleted."""
+
+    fake_backend = mocker.Mock(spec=testinfra.backend.base.BaseBackend)
+    myhost = testinfra.host.Host(fake_backend)
+    cr1 = mocker.Mock(spec=testinfra.backend.base.CommandResult)
+    cr2 = mocker.Mock(spec=testinfra.backend.base.CommandResult)
+    mocker.patch('testinfra.host.Host.run', side_effect=[cr1, cr2, cr2])
```
Comment (zreichert) [inline tests/helpers/test_delete_instance.py]: I'm not sure I understand where `[cr1, cr2, cr2]` are being used. I see one of them ( the return from .run())

Comment (zreichert) [reply tests/helpers/test_delete_instance.py]: > If side_effect is an iterable then each call to the mock will return the next value from the iterable.
> - Python docs

OK, I think I just answered my own question. other methods in this helper will call `.run()`

Comment (zreichert) [reply tests/helpers/test_delete_instance.py]: Neat trick!

Comment (johnduarte) [reply tests/helpers/test_delete_instance.py]: The first call to `run` is triggered by the [get_id_by_name](https://github.com/rcbops/pytest-rpc/blob/master/pytest_rpc/helpers.py#L233) call and should return a CommandResult that includes the json with the server name and id. The second (and subsequent) call to `run` is triggered by the [asset_not_in_list](https://github.com/rcbops/pytest-rpc/blob/master/pytest_rpc/helpers.py#L238) call and should return CommandResult that does not include the server.

**Pull request merged**